File: CS/09_Forms/FillFormField.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Fields;
using Spire.Pdf.Widget;

namespace FillFormField
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        { 
            //Load a pdf document
            string input = @"..\..\..\..\..\..\Data\FillFormField.pdf";
            PdfDocument doc = new PdfDocument();
	    doc.LoadFromFile(input);
            //Get pdf forms
            PdfFormWidget formWidget = doc.Form as PdfFormWidget;

            //Fill pdf form fields
            for (int i = 0; i < formWidget.FieldsWidget.List.Count; i++)
            {
                PdfField field = formWidget.FieldsWidget.List[i] as PdfField;
                if (field is PdfTextBoxFieldWidget)
                {
                    PdfTextBoxFieldWidget textBoxField = field as PdfTextBoxFieldWidget;
                    switch (textBoxField.Name)
                    {
                        case "email":
                            textBoxField.Text = "support@e-iceblue.com";
                            break;

                        case "username":
                            textBoxField.Text = "E-iceblue";
                            break;

                        case "password":
                            textBoxField.Password = true;
                            textBoxField.Text = "e-iceblue";
                            break;

                        case "password2":
                            textBoxField.Password = true;
                            textBoxField.Text = "e-iceblue";
                            break;

                        case "company_name ":
                            textBoxField.Text = "E-iceblue";
                            break;

                        case "first_name":
                            textBoxField.Text = "James";
                            break;

                        case "last_name":
                            textBoxField.Text = "Chen";
                            break;

                        case "middle_name":
                            textBoxField.Text = "J";
                            break;

                        case "address1":
                            textBoxField.Text = "Chengdu";
                            break;

                        case "address2":
                            textBoxField.Text = "Beijing";
                            break;

                        case "city":
                            textBoxField.Text = "Shanghai";
                            break;

                        case "postal_code":
                            textBoxField.Text = "11111";
                            break;

                        case "state":
                            textBoxField.Text = "Shanghai";
                            break;

                        case "phone":
                            textBoxField.Text = "1234567901";
                            break;

                        case "mobile_phone":
                            textBoxField.Text = "123456789";
                            break;

                        case "fax":
                            textBoxField.Text = "12121212";
                            break;
                    }
                }

                if (field is PdfListBoxWidgetFieldWidget)
                {
                    PdfListBoxWidgetFieldWidget listBoxField = field as PdfListBoxWidgetFieldWidget;
                    switch (listBoxField.Name)
                    {
                        case "email_format":
                            int[] index = { 1 };
                            listBoxField.SelectedIndex = index;
                            break;
                    }
                }

                if (field is PdfComboBoxWidgetFieldWidget)
                {
                    PdfComboBoxWidgetFieldWidget comBoxField = field as PdfComboBoxWidgetFieldWidget;
                    switch (comBoxField.Name)
                    {
                        case "title":
                            int[] items = { 0 };
                            comBoxField.SelectedIndex = items;
                            break;
                    }
                }

                if (field is PdfRadioButtonListFieldWidget)
                {
                    PdfRadioButtonListFieldWidget radioBtnField = field as PdfRadioButtonListFieldWidget;
                    switch (radioBtnField.Name)
                    {
                        case "country":
                            radioBtnField.SelectedIndex = 1;
                            break;
                    }
                }

                if (field is PdfCheckBoxWidgetFieldWidget)
                {
                    PdfCheckBoxWidgetFieldWidget checkBoxField = field as PdfCheckBoxWidgetFieldWidget;
                    switch (checkBoxField.Name)
                    {
                        case "agreement_of_terms":
                            checkBoxField.Checked = true;
                            break;
                    }
                }

                if (field is PdfButtonWidgetFieldWidget)
                {
                    PdfButtonWidgetFieldWidget btnField = field as PdfButtonWidgetFieldWidget;
                    switch (btnField.Name)
                    {

                        case "submit":
                            btnField.Text = "Submit";
                            break;
                    }
                }
            }

            string output = "FillFormField.pdf";

            //Save pdf document
            doc.SaveToFile(output);

            //Launch the file
            PDFDocumentViewer(output);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB/09_Forms/FillFormField.vb
Imports Spire.Pdf
Imports Spire.Pdf.Fields
Imports Spire.Pdf.Widget

Namespace FillFormField
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Define the input file path
            Dim input As String = "..\..\..\..\..\..\Data\FillFormField.pdf"

            ' Create a new PdfDocument object
            Dim doc As New PdfDocument()

            ' Load the PDF document from the specified input file
            doc.LoadFromFile(input)

            ' Get the form widget from the loaded PDF document
            Dim formWidget As PdfFormWidget = TryCast(doc.Form, PdfFormWidget)

            ' Iterate through each field in the form
            For i As Integer = 0 To formWidget.FieldsWidget.List.Count - 1
                ' Get the current field
                Dim field As PdfField = TryCast(formWidget.FieldsWidget.List(i), PdfField)

                ' Check if the field is a TextBox field
                If TypeOf field Is PdfTextBoxFieldWidget Then
                    ' Cast the field as a TextBox field
                    Dim textBoxField As PdfTextBoxFieldWidget = TryCast(field, PdfTextBoxFieldWidget)

                    ' Fill the TextBox fields based on their names
                    Select Case textBoxField.Name
                        Case "email"
                            textBoxField.Text = "support@e-iceblue.com"

                        Case "username"
                            textBoxField.Text = "E-iceblue"

                        Case "password"
                            textBoxField.Password = True
                            textBoxField.Text = "e-iceblue"

                        Case "password2"
                            textBoxField.Password = True
                            textBoxField.Text = "e-iceblue"

                        Case "company_name"
                            textBoxField.Text = "E-iceblue"

                        Case "first_name"
                            textBoxField.Text = "James"

                        Case "last_name"
                            textBoxField.Text = "Chen"

                        Case "middle_name"
                            textBoxField.Text = "J"

                        Case "address1"
                            textBoxField.Text = "Chengdu"

                        Case "address2"
                            textBoxField.Text = "Beijing"

                        Case "city"
                            textBoxField.Text = "Shanghai"

                        Case "postal_code"
                            textBoxField.Text = "11111"

                        Case "state"
                            textBoxField.Text = "Shanghai"

                        Case "phone"
                            textBoxField.Text = "1234567901"

                        Case "mobile_phone"
                            textBoxField.Text = "123456789"

                        Case "fax"
                            textBoxField.Text = "12121212"
                    End Select
                End If

                ' Check if the field is a ListBox field
                If TypeOf field Is PdfListBoxWidgetFieldWidget Then
                    ' Cast the field as a ListBox field
                    Dim listBoxField As PdfListBoxWidgetFieldWidget = TryCast(field, PdfListBoxWidgetFieldWidget)

                    ' Fill the ListBox fields based on their names
                    Select Case listBoxField.Name
                        Case "email_format"
                            Dim index() As Integer = {1}
                            listBoxField.SelectedIndex = index
                    End Select
                End If

                ' Check if the field is a ComboBox field
                If TypeOf field Is PdfComboBoxWidgetFieldWidget Then
                    ' Cast the field as a ComboBox field
                    Dim comBoxField As PdfComboBoxWidgetFieldWidget = TryCast(field, PdfComboBoxWidgetFieldWidget)

                    ' Fill the ComboBox fields based on their names
                    Select Case comBoxField.Name
                        Case "title"
                            Dim items() As Integer = {0}
                            comBoxField.SelectedIndex = items
                    End Select
                End If

                ' Check if the field is a RadioButtonList field
                If TypeOf field Is PdfRadioButtonListFieldWidget Then
                    ' Cast the field as a RadioButtonList field
                    Dim radioBtnField As PdfRadioButtonListFieldWidget = TryCast(field, PdfRadioButtonListFieldWidget)

                    ' Fill the RadioButtonList fields based on their names
                    Select Case radioBtnField.Name
                        Case "country"
                            radioBtnField.SelectedIndex = 1
                    End Select
                End If

                ' Check if the field is a CheckBox field
                If TypeOf field Is PdfCheckBoxWidgetFieldWidget Then
                    ' Cast the field as a CheckBox field
                    Dim checkBoxField As PdfCheckBoxWidgetFieldWidget = TryCast(field, PdfCheckBoxWidgetFieldWidget)

                    ' Fill the CheckBox fields based on their names
                    Select Case checkBoxField.Name
                        Case "agreement_of_terms"
                            checkBoxField.Checked = True
                    End Select
                End If

                ' Check if the field is a Button field
                If TypeOf field Is PdfButtonWidgetFieldWidget Then
                    ' Cast the field as a Button field
                    Dim btnField As PdfButtonWidgetFieldWidget = TryCast(field, PdfButtonWidgetFieldWidget)

                    ' Fill the Button fields based on their names
                    Select Case btnField.Name
                        Case "submit"
                            btnField.Text = "Submit"
                    End Select
                End If
            Next i

            ' Define the output file path
            Dim output As String = "FillFormField.pdf"

            'Save the document to the file path
            doc.SaveToFile(output)

            ' Close the document
            doc.Close()

            ' Launch the file
            PDFDocumentViewer(output)
        End Sub
        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try
        End Sub
    End Class
End Namespace
